src/combat.ts

```diff
@@ -134,12 +134,49 @@ function shouldRedigitize() {
   return SourceTerminal.have() && myAdventures() * 1.04 < digitizesLeft * digitizeAdventuresUsed;
 }
 
+type Constructor<T> = { new (): T };
+
 export class Macro extends LibramMacro {
   submit(): string {
     print(this.components.join("\n"));
     return super.submit();
   }
 
+  skill(...skills: Skill[]): this {
+    return super.skill(...skills);
+  }
+
+  static skill<T extends LibramMacro>(this: Constructor<T>, ...skills: Skill[]): T {
```

Comment:
should be `T extends Macro`, no?

Reply:
It seems like, for whatever reason, VSCode hates that (red squigglies), but I can do it in my libram PR. I'd be willing to guess the issue is recycling the name 'Macro.' I'll close this PR and wait until we get StrictMacro in libram.